jsonstring.go

```diff
@@ -98,7 +98,7 @@ func writeQuotedBytes(w io.Writer, s []byte) (int, error) {
 	start := 0
 	for i := 0; i < len(s); {
 		if b := s[i]; b < utf8.RuneSelf {
-			if 0x20 <= b && b != '\\' && b != '"' {
+			if 0x20 <= b && b != '\\' && b != '"' && b != 0x7f {
```

Comment:
Per the NOTE above, the `writeQuotedString` function above needs the same additional check.

That also means we should add a test case for the string version of `"\x7f"` in `encode_test.go` near the new test case added in the previous PR.